pkg/resources/kafka/kafka.go

```diff
@@ -928,11 +928,15 @@ func (r *Reconciler) getK8sAssignedNodeport(log logr.Logger, eListenerName strin
 
 	var nodePort int32
 	for _, port := range nodePortSvc.Spec.Ports {
-		if port.Name == fmt.Sprintf("broker-%d", brokerId) {
+		if port.Name == fmt.Sprintf("broker-%d", brokerId) && port.NodePort != 0 {
```

Comment:
`&& port.NodePort != 0` is this needed? 
`if nodePort == 0 ` below will handle the zero case anyway